app/controllers/invoice_controller.rb

```diff
@@ -0,0 +1,43 @@
+require 'net/http'
+require 'uri'
+class InvoiceController < ApplicationController
+  def index
+  end
+
+  def show
+    invoice_db = InvoiceDB.new
+    service = RetreiveClientService.new(invoice_db)
```

Comment:
Are you sure? I really liked it. We use this kind of indirections on IX, but call them interactors and gateways.
https://blog.8thlight.com/uncle-bob/2012/08/13/the-clean-architecture.html